nimare/workflows/macm.py

```diff
@@ -1,12 +1,157 @@
-                   if k.startswith('kernel__')}
-    kwargs = {k: v for k, v in kwargs.items() if not k.startswith('kernel__')}
-    del kernel_args
-    del kwargs
+"""
+
+"""
+import os
+import pathlib
+from shutil import copyfile
+
+import click
+import numpy as np
+
+from ..dataset import Dataset
+from ..meta.cbma import ALE
+
+N_ITERS_DEFAULT = 10000
+CLUSTER_FORMING_THRESHOLD_P_DEFAULT = 0.001
+CLUSTER_SIZE_THRESHOLD_Q_DEFAULT = 0.05
+
+
+@click.command(name='macm',
+               short_help='Run a meta-analytic coactivation modeling (MACM) '
+                          'analysis using activation likelihood estimation '
+                          '(ALE) on a NiMARE dataset file and a target mask.')
+@click.argument('dataset_file', type=click.Path(exists=True))
+@click.option('--mask', '--mask_file', type=click.Path(exists=True))
+@click.option('--output_dir', help="Where to put the output maps.")
+@click.option('--prefix', help="Common prefix for output maps.")
+@click.option('--n_iters', default=N_ITERS_DEFAULT, show_default=True,
+              help="Number of iterations for permutation testing.")
+@click.option('--v_thr', default=CLUSTER_FORMING_THRESHOLD_P_DEFAULT,
+              show_default=True,
+              help="Voxel p-value threshold used to create clusters.")
+@click.option('--c_thr', default=CLUSTER_SIZE_THRESHOLD_Q_DEFAULT,
+              show_default=True,
+              help="Cluster size corrected p-value threshold.")
+@click.option('--n_cores', default=-1,
+              show_default=True,
+              help="Number of processes to use for meta-analysis. If -1, use "
+                   "all available cores.")
+def macm_workflow(dataset_file, mask_file, output_dir=None,
+                  prefix=None,
+                  n_iters=N_ITERS_DEFAULT,
+                  v_thr=CLUSTER_FORMING_THRESHOLD_P_DEFAULT,
+                  c_thr=CLUSTER_SIZE_THRESHOLD_Q_DEFAULT,
+                  n_cores=-1):
+    """
+    Perform MACM with ALE algorithm.
+    """
+    click.echo("Loading coordinates...")
+    dset = Dataset(dataset_file)
+    sel_ids = dset.get_studies_by_mask(mask_file)
+
+    # override sample size
+    n_subs_db = dset.coordinates.drop_duplicates('id')['n'].astype(float).astype(int).sum()
+    sel_coords = dset.coordinates.loc[dset.coordinates['id'].isin(sel_ids)]
+    n_subs_sel = sel_coords.drop_duplicates('id')['n'].astype(float).astype(int).sum()
+    click.echo("{0} studies selected out of {1}.".format(len(sel_ids),
+                                                         len(dset.ids)))
+
+    boilerplate = """
+Meta-analytic connectivity modeling (MACM; Laird et al., 2009; Robinson et al.,
+2009; Eickhoff et al., 2010) analysis was performed with the activation
+likelihood estimation (ALE; Turkeltaub, Eden, Jones, & Zeffiro, 2002; Eickhoff,
+Bzdok, Laird, Kurth, & Fox, 2012; Turkeltaub et al., 2012) meta-analysis
+algorithm using NiMARE. The input database included {n_foci_db}
```

Comment:
dataset?

Reply:
I went with database here instead of dataset since studies are being selected from within it, rather than running the full thing. Do you prefer dataset?